app/src/main/java/com/example/shareyourtrip/PostAdapter.java

```diff
@@ -89,15 +137,67 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                     //If favorite is toggled on...
                     if (isChecked) {
                         holder.favButton.setBackgroundResource(R.drawable.ic_favorite);
-                        //todo: Put favorite database stuff here
+
+                        if( (currentContext instanceof SearchActivity) || (currentContext instanceof HomePageActivity) ) {
+                            //todo: Put favorite database stuff here
+                            int pos = holder.getAdapterPosition();
+
+                            // check if item still exists
+                            if (pos != RecyclerView.NO_POSITION) {
+                                Post post = postList.get(pos);
+                                clickedPost = new Post(post);
+
+                                if (!postDAO.insertFavPost(post)) {
+                                    Toast.makeText(currentContext, "This post is already you favortie!", Toast.LENGTH_LONG).show();
+                                }
+                            }
+                        }
```

Comment:
What if the favorite button is toggled on it the favorite activity? Would need "else if(currentContext == Favorite Activity"